src/collector/worker.rs

```diff
@@ -0,0 +1,150 @@
+    unsafe { PgRefCell::<Option<rusqlite::Connection>>::new(None) };
+
+fn get<'a>() -> Option<RefMut<'a, rusqlite::Connection>> {
+    if unsafe { !pgrx::pg_sys::IsUnderPostmaster } {
+        return None;
+    }
+    let database_oid = unsafe { pgrx::pg_sys::MyDatabaseId.to_u32() };
+    if database_oid == 0 {
+        return None;
+    }
+    let mut connection = CONNECTION.borrow_mut();
+    if connection.is_none() {
+        if !Path::new(COLLECTOR_DIR).exists() {
+            fs::create_dir_all(COLLECTOR_DIR).ok()?;
+        }
+        let p = format!("pg_vchord_sample/database_{database_oid}.sqlite");
+        let conn = rusqlite::Connection::open(&p).ok()?;
+        let version: u32 = conn
+            .pragma_query_value(Some("schema"), "user_version", |row| row.get(0))
+            .unwrap_or(COLLECTOR_VERSION);
+        let conn = if version != COLLECTOR_VERSION && version != 0 {
+            drop(conn);
+            fs::remove_file(&p).ok()?;
+            rusqlite::Connection::open(&p).ok()?
+        } else {
+            conn
+        };
+        // SET journal_mode=WAL for read/write concurrency
+        conn.pragma_update(Some("main"), "journal_mode", "WAL")
+            .ok()?;
+        conn.pragma_update(Some("main"), "user_version", COLLECTOR_VERSION)
+            .ok()?;
+        connection.replace(conn);
+    }
+    if connection.is_none() {
+        return None;
+    }
+    Some(RefMut::map(connection, |c| c.as_mut().unwrap()))
+}
+
+pub fn push(index_oid: u32, sample: &str, max_records: u32) {
+    let mut connection = match get() {
+        Some(c) => c,
+        None => return,
+    };
+    let init_statement = format!(
+        "
+        CREATE TABLE IF NOT EXISTS index_{index_oid} (id INTEGER PRIMARY KEY, sample TEXT, create_at REAL);
+        CREATE INDEX IF NOT EXISTS i ON index_{index_oid} (create_at DESC, id DESC);
+        "
+    );
+    let insert_statement = format!(
+        "INSERT INTO index_{index_oid} (sample, create_at) VALUES (?1, unixepoch('subsec'))"
+    );
+    let maintain_statement = format!(
+        "DELETE FROM index_{index_oid} WHERE (create_at, id) < (
+            SELECT create_at, id FROM index_{index_oid}
+            ORDER BY create_at DESC, id DESC
+            LIMIT 1 OFFSET ({max_records} - 1))"
+    );
+    let tx = match connection.transaction() {
+        Ok(t) => t,
+        Err(e) => {
+            pgrx::debug1!("Collector: Error starting transaction: {}", e);
+            return;
+        }
+    };
+    if let Err(e) = tx.execute_batch(&init_statement) {
+        pgrx::debug1!("Collector: Error creating table: {}", e);
+        return;
+    };
+    if let Err(e) = tx.execute(&insert_statement, (sample,)) {
+        pgrx::debug1!("Collector: Error inserting sample: {}", e);
+        return;
+    }
+    if let Err(e) = tx.execute(&maintain_statement, ()) {
+        pgrx::debug1!("Collector: Error maintaining samples: {}", e);
+        return;
+    }
+    let _ = tx.commit();
```

Comment:
Extract the code into an immediately invoked closure, and ignore the returned `Err` or print logs.